pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -130,7 +130,10 @@ func (a *index) serviceEntriesInfo(s *networkingclient.ServiceEntry, w *Waypoint
 	waypoint := model.WaypointBindingStatus{}
 	if w != nil {
 		waypoint.ResourceName = w.ResourceName()
-		waypoint.IngressUseWaypoint = s.Labels["istio.io/ingress-use-waypoint"] == "true"
+		if val, ok := s.Labels["istio.io/ingress-use-waypoint"]; ok {
+			waypoint.IngressLabelPresent = true
+			waypoint.IngressUseWaypoint = strings.EqualFold(val, "true")
```

Comment:
In EndpointBuilder.findServiceWaypoint, you compare considering case sensitive

Reply:
hmm where do you see that? I just see `!svc.IngressUseWaypoint`

Reply:
```
func (b *EndpointBuilder) findServiceWaypoint(endpointIndex *model.EndpointIndex) ([]*model.IstioEndpoint, bool) {
	if b.nodeType != model.Router {
		// Currently only ingress will call waypoints
		return nil, false
	}
	// ...and they must explicitly opt in
	if b.service.Attributes.Labels["istio.io/ingress-use-waypoint"] != "true" {
		return nil, false
	}
```

Reply:
I think you are looking at an old commit?

Reply:
ah, right

Reply:
Another place 

```
	NamespacesInfo := krt.NewCollection(Namespaces, func(ctx krt.HandlerContext, i *v1.Namespace) *model.NamespaceInfo {
		return &model.NamespaceInfo{
			Name:               i.Name,
			IngressUseWaypoint: i.Labels["istio.io/ingress-use-waypoint"] == "true",
		}
	}, opts.WithName("NamespacesInfo")...)
```